modules/context/auth.go

```diff
@@ -12,6 +12,7 @@ import (
 	"code.gitea.io/gitea/modules/log"
 	"code.gitea.io/gitea/modules/setting"
 	"code.gitea.io/gitea/modules/web/middleware"
+	"code.gitea.io/gitea/services/audit"
```

Comment:
Not a fan.
At all.
We are trying to get rid of `services` imports in `modules`, not add them!

Reply:
I guess we need to refactor the Gitea's `Context` slightly to decouple some dependencies.

Reply:
`auth` can't be a module because it must use models. So by definition it must be a service and then it's alright to use another service. These two methods should be part of `services/auth/middleware.go`.

Reply:
Added a PR for that: #24287